apps/full-site-editing/full-site-editing-plugin/site-editor-experiment/index.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * Site Editor experiment file.
+ *
+ * @package A8C\FSE
+ */
+
+namespace A8C\FSE;
+
+/**
+ * Enables/Disables site editor experiment per blog sticker.
+ */
+function conditionally_enable_site_editor() {
+	// Check blog sticker for access to Site Editor.
+	if ( ! has_blog_sticker( 'core_site_editor_enabled' ) ) {
+
+		// Check if experiment is enabled: disable if needed.
+		if ( gutenberg_is_experiment_enabled( 'gutenberg-full-site-editing' ) ) {
+			set_site_editor_experiment( '0' );
+		}
+		return;
+	}
+
+	// Check if experiment is disabled: enable if needed.
+	if ( ! gutenberg_is_experiment_enabled( 'gutenberg-full-site-editing' ) ) {
+		set_site_editor_experiment( '1' );
+	}
+
+	add_site_editor_plugin_page();
+	override_site_editor_init();
+}
+
+/**
+ * Sets the Site Editor experiment on or off.
+ *
+ * @param {string} $value - '0' for OFF, '1' for ON.
+ */
+function set_site_editor_experiment( $value ) {
+	$options = get_option( 'gutenberg-experiments' );
+
+	// Initialize options array if needed (may be set to false by default on dotcom).
+	if ( ! $options ) {
+		$options = array();
+	}
+	$options['gutenberg-full-site-editing']      = $value;
+	$options['gutenberg-full-site-editing-demo'] = $value;
+	update_option( 'gutenberg-experiments', $options );
+}
+
+/**
+ * Adds Site Editor page to wp-admin sidebar.
+ */
+function add_site_editor_plugin_page() {
+	// Without overriding init, this would need to be a submenu of gutenberg.
+	add_menu_page(
+		__( 'Site Editor (beta)', 'gutenberg' ),
+		__( 'Site Editor (beta)', 'gutenberg' ),
+		'edit_theme_options',
+		'gutenberg-edit-site',
+		'gutenberg_edit_site_page',
+		'dashicons-edit'
+	);
+}
+
+/**
+ * Initialize the Gutenberg Edit Site Page.
+ * Copy of core function from 7.3.0 - gutenberg_edit_site_init
+ *
+ * Currently overriding to change hook requirement to allow Site Editor at top level sidebar.
+ * And to troubleshoot the wp.editSite.initialize CORS error.
+ *
+ * Any changes will be prefixed with @DOTCOM_CUSTOMIZATION comment.
+ *
+ * @param string $hook Page.
+ */
+function fse_plugin_edit_site_init( $hook ) {
```

Comment:
We probably don't want to re-write this entire function in the end.  This is currently a copy of core's 'gutenberg_edit_site_init' function (https://github.com/WordPress/gutenberg/blob/e01b562f6e87a5a3ed73282cc71f28dee0712ae0/lib/edit-site-page.php#L30) on the version we are currently running for dotcom.  To keep this would probably mean we would need to update this function every time dotcom upgrades gutenberg versions and this file is changed.

This was currently necessary to:
1. Change the $hook accepted by the early return statement to allow our 'Site Editor' button to be at the top-level of the sidebar.
2. Trigger the `enqueue_block_editor_assets` action to allow things like apiFetch to be properly set up for dotcom.